### PR TITLE
Add conditional compilation flags for anything using blake2

### LINKED ISSUE #10: Introduce conditional compilation for BLS12_381 signatures
This should be done in order to give the option to only pull the data structure from the library and allow for dusk-network/rusk#148 to properly compile without std inclusion for hashing libraries

### PR DESCRIPTION
**Pull request by julesdesmit**
Fixes #10

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,10 +9,10 @@ canonical = { version = "0.5", optional = true }
 canonical_derive = {version = "0.5", optional = true}
 dusk-bls12_381 = { version = "0.6", default-features = false }
```
Comment (vlopes11) [inline Cargo.toml:10]: You need to use the feature `groups` to use affine structs

**Review by vlopes11 — CHANGES_REQUESTED** [review]
Small nit

File: Cargo.toml (hunk 1 above)
Comment (julesdesmit) [reply Cargo.toml:10]: The desired features are imported under `[features]` in the toml

Comment (vlopes11) [reply Cargo.toml:10]: You use `groups` both for `std` and `no-std`.

This line should be:
`dusk-bls12_381 = { version = "0.6", default-features = false, features=["groups"] }`

**Review by vlopes11 — APPROVED** [review]
LGTM

**Pull request merged**